Updating loss calculation to use one_hot labels

dandouthit opened the pull request:
The in-depth description for the code uses one_hot labels as input to the softmax_cross_entropy() function, therefore the code should use one hot labels as well.

dandouthit committed:
Updating loss calculation to use one_hot vectors

The in-depth description for the code uses one_hot labels as input to the softmax_cross_entropy() function, therefore the code should use one hot labels as well.

MarkDaoust reviewed (changes requested):
Hi,

Thanks for pointing out, and taking the time to fix this inconsistency.

I actually really prefer the "sparse_" version (more efficient, less code, no "please use `tf.nn.softmax_cross_entropy_with_logits_v2` warning.").

Any chance you could change the direction here and fix the tutorial to match the code instead?

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/docs_src/tutorials/layers.md

Thanks.

dandouthit commented:
@MarkDaoust Sure thing, I'll keep the sparse_softmax_cross_entropy() version of the code, and update the tutorial to match. Changes to come shortly.

dandouthit committed:
Updating tutorial to use sparse_softmax_cross_entropy()

As per request from @MarkDaoust, I have updated the tutorial section of the notebook to reflect the usage of tf.losses.sparse_softmax_cross_entropy() rather than tf.losses.softmax_cross_entropy() using one-hot vectors.

dandouthit commented:
@MarkDaoust Were my changes in the second commit d7c971c1563cb61f0df6dc6ce3c89641ca0a5d8a sufficient for your review, or was there anything further that you'd like me to address?

MarkDaoust committed:
More simplifications to the text.
